express-api/src/services/notifications/notificationServices.ts

```diff
@@ -0,0 +1,206 @@
+    To: to ?? template.To,
+    Body: body,
+    TemplateId: template.Id,
+  };
+  return AppDataSource.getRepository(NotificationQueue).save(queueObject);
+};
+
+const insertProjectNotificationQueue = async (
+  template: NotificationTemplate,
+  projStatusNotif: ProjectStatusNotification,
+  project: Project,
+  agency?: Agency,
+  overrideTo?: string,
+) => {
+  const sendDate = new Date();
+  sendDate.setDate(sendDate.getDate() + projStatusNotif.DelayDays);
+  const queueObject = {
+    Key: randomUUID(),
+    Status: NotificationStatus.Pending,
+    Priority: template.Priority,
+    Encoding: template.Encoding,
+    SendOn: sendDate,
+    Subject: nunjucks.renderString(template.Subject, { Project: project }),
+    Body: nunjucks.renderString(template.Body, {
+      Title: Title,
+      Uri: Uri,
+      ToAgency: agency,
+      Project: flattenProjectProperties(project),
+    }),
+    TemplateId: template.Id,
+    To: [overrideTo ?? agency?.Email, template.To].filter((a) => a).join(';'),
+    Cc: [agency?.CCEmail, template.Cc].filter((a) => a).join(';'),
+    Bcc: template.Bcc,
+  };
+  const insertedNotif = await AppDataSource.getRepository(NotificationQueue).save(queueObject);
+  return insertedNotif;
+};
+
+const generateProjectNotifications = async (project: Project, previousStatusId: number) => {
+  const projectStatusNotif1 = await AppDataSource.getRepository(ProjectStatusNotification).find({
+    where: { FromStatusId: previousStatusId, ToStatusId: project.StatusId },
+  });
+  const projectStatusNotif2 = await AppDataSource.getRepository(ProjectStatusNotification).find({
+    where: { FromStatusId: IsNull(), ToStatusId: project.StatusId },
+  });
+  const projectStatusNotifications = [...projectStatusNotif1, ...projectStatusNotif2];
+  const returnNotifications = [];
+  for (const projStatusNotif of projectStatusNotifications) {
+    const template = await AppDataSource.getRepository(NotificationTemplate).findOne({
+      where: { Id: projStatusNotif.Id },
+    });
+
+    const sendDate = new Date();
+    sendDate.setDate(sendDate.getDate() + projStatusNotif.DelayDays);
+
+    let overrideTo: string | null = null;
+    if (template.Audience == NotificationAudience.ProjectOwner) {
+      const owningUser = await AppDataSource.getRepository(User).findOne({
+        where: { Id: project.CreatedById },
+      });
+      overrideTo = owningUser.Email;
+      returnNotifications.push(
+        insertProjectNotificationQueue(
+          template,
+          projStatusNotif,
+          project,
+          project.Agency,
+          overrideTo,
+        ),
+      );
+    } else if (template.Audience == NotificationAudience.OwningAgency) {
+      returnNotifications.push(
+        insertProjectNotificationQueue(template, projStatusNotif, project, project.Agency),
+      );
+    } else if (template.Audience == NotificationAudience.Agencies) {
+      //Dunno if we really have/plan to have the ProjectAgencyResponses implemented?
```

Comment:
Currently, Josh, manually adds an agency's interest within an ERP project if an agency expresses interest, which will then trigger 3 emails for the 30/60/90 notification being sent to CHES for this agency which wanted to be "notified".

Since we don't have any UI designs for this, we'll probably need to bring Aiden and Josh into this conversation to see what's needed.

In the C# GenerateNotifciations service, there is a check for the template audience where if it's parent agencies, it will get all the emails for each of these to generate a notification to send.

I believe the majority of this content is still relevant: https://github.com/bcgov/PIMS/wiki/Notifications